levelkdev/swapr-dapp
# Fix/injected connector

# Summary
Fixes #624 #683 #699 

*Bug Description*
Activation of the Injected Connector (which is used with MetaMask and Coinbase wallets) results in displaying pop-ups from both wallets at the same time when the user has both extensions installed. Activation of the Caoinbase wallet doesn't cause the problem.

More details can be found as the description of solved issues mentioned above.


  # To Test

1. Install MetaMask and Coinbase Wallet chrome extensions.
- [ ] Button shows 'connect wallet'
- [ ] Non of the wallet is connected automatically

2. Click the "Connect Wallet" button.
- [ ] If MetaMask is selected only MetaMask popup should be displayed
- [ ] If Coinbase is selected only Coinbase popup should be displayed

3. Proceed with connection process until wallet is Connected and address can be seen
- [ ] the button in upper right corner and eg bridge button works as expected and open wallet list to be selected
![image](https://user-images.githubusercontent.com/88723742/160019714-55de0d06-b72c-4c29-ad83-e002d4c1c0fd.png)

- [ ] connect manually with unsupported network (eg Kovan) - switching from unsupported to supported network (eg. Mainnet) works as expected
- [ ] try any transaction, like swap - works as expected and correct wallet popup is shown

4. Click the "Disconnect Wallet" button and connect with different one.
- [ ] only one popup appears and allows for connection 
- [ ] switching networks and transactions flow work as expected
- [ ] **important! addresses are different after wallet switching and matches to the wallets addresses appropriately** 
![image](https://user-images.githubusercontent.com/88723742/160021573-a5625528-8b0d-47ca-a182-4a97fcf2aa9b.png)
![image](https://user-images.githubusercontent.com/88723742/160021814-85bad502-48ba-402a-b426-1bccdc425164.png)


5. Disconnect wallet manually. This HAS TO BE DONE MANUALLY and it's not the same as "Disconnect wallet" on our app (this is for purpose and it's not the bug!). To disconnect manually use Metamask and Coinbase wallet app as shown below.
Try connect to the wallet again and Reject transaction.
![image](https://user-images.githubusercontent.com/88723742/160020855-8e5d6842-632f-4cc0-b356-c56fb9b0fd6c.png)
![image](https://user-images.githubusercontent.com/88723742/160020975-abd0beed-127f-496d-90d1-37dfc1ae6b43.png)


- [ ] Error modal should be shown
![image](https://user-images.githubusercontent.com/88723742/160021084-722b51e5-890c-4c1e-9676-f9e5dae43674.png)

- [ ] "Try again" button works as expected and allow to try again
- [ ] Closing modal and trying to activate another wallet works as expected

6. Any "randomly" usage the app and wallets to test edge cases will be appreciated :)
